pkg/console/controllers/route/controller.go

```diff
@@ -126,21 +132,51 @@ func (c *RouteSyncController) sync() error {
 		}
 		return c.removeRoute(api.OpenShiftConsoleName)
 	default:
-		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+		return fmt.Errorf("unknown state: %v", updatedOperatorConfig.Spec.ManagementState)
 	}
 
-	updatedOperatorConfig := operatorConfig.DeepCopy()
+	conditionFuncs := []v1helpers.UpdateStatusFunc{}
```

Comment:
So can we do something like this in `status.go`:

```golang
type conditionHandler struct {
	client   v1helpers.OperatorClient
	updaters []v1helpers.UpdateStatusFunc
}

func (c conditionHandler) Add(updater v1helpers.UpdateStatusFunc) {
	c.updaters = append(c.updaters, updater)
}

func (c conditionHandler) Flush() error {
        // though, we really need to call all the funcs, catch the errors, then clear 
       // the updaters to an empty slice before next use: 
       // c.updaters = []v1helpers.UpdateStatusFunc
	if _, _, updateErr := v1helpers.UpdateStatus(c.client, c.updaters...); updateErr != nil {
		return updateErr
	}
	return nil 
}

func NewConditionHandler(client v1helpers.OperatorClient) conditionHandler {
	return conditionHandler{
		client: client,
	}
}
```
and then here do something like:

```golang
        // we could call this in `.sync()`, which would let us just throw it away at the end of 
        // each sync loop and not worry about clearing out the slice of []v1helpers.UpdateStatusFunc.
       // alternatively we create these and attach them to our controllers when we make RouteSyncController{} 
       // in NewRouteSyncController() func (and with all of our other controllers)...  but that means we defintely
       // have to have the clearing logic work to ensure a func doesn't linger across the .sync() loop calls. 
	conditionsHandler := status.NewConditionHandler(c.operatorClient)

	defaultRoute, defaultRouteErrReason, defaultRouteErr := c.SyncDefaultRoute(updatedOperatorConfig)
	defaultRouteSyncConditionsFunc := status.HandleProgressingOrDegraded("DefaultRouteSync", defaultRouteErrReason, defaultRouteErr)
	conditionsHandler.Add(defaultRouteSyncConditionsFunc)
	if defaultRouteErr != nil {
                // i guess this can return an error, also, and we need to deal.  ideally 
                // we clobber the nested if statements
		conditionsHandler.Flush() 
		return defaultRouteErr
	}
```